src/AWS.Deploy.Common/Recommendation.cs

```diff
@@ -200,10 +200,35 @@ public bool IsOptionSettingDisplayable(OptionSettingItem optionSetting)
                 var dependsOnOptionSettingValue = GetOptionSettingValue(dependsOnOptionSetting);
                 if (
                     dependsOnOptionSetting != null &&
-                    dependsOnOptionSettingValue != null &&
-                    !dependsOnOptionSettingValue.Equals(dependency.Value))
+                    dependsOnOptionSettingValue != null)
                 {
-                    return false;
+                    if (dependency.Operation == null ||
+                           dependency.Operation == PropertyDependencyOperationType.Equals)
+                    {
+                        if (!dependsOnOptionSettingValue.Equals(dependency.Value))
+                            return false;
+                    }
+                    else if (dependency.Operation == PropertyDependencyOperationType.NotEmpty)
+                    {
+                        if (dependsOnOptionSetting.Type == OptionSettingValueType.List &&
+                            dependsOnOptionSettingValue != null &&
+                            dependsOnOptionSettingValue.TryDeserialize<SortedSet<string>>(out var listValue) &&
+                            !listValue.Any())
+                        {
+                            return false;
+                        }
+                        else if (dependsOnOptionSetting.Type == OptionSettingValueType.KeyValue &&
+                            dependsOnOptionSettingValue != null &&

```

Comment:
Same comment as above about null value for `dependsOnOptionSettingValue`.

Reply:
The condition on lineThe condition on line 203 https://github.com/aws/aws-dotnet-deploy/blob/842ae771dc31933231b30fe19abcadc269ca3534/src/AWS.Deploy.Common/Recommendation.cs#L203 makes sure that `dependsOnOptionSettingValue` is not null. In any case, `dependsOnOptionSettingValue` is never null because of https://github.com/aws/aws-dotnet-deploy/blob/842ae771dc31933231b30fe19abcadc269ca3534/src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs#L53 
`dependsOnOptionSettingValue` will be an empty string which will be handled in the if condition.

Reply:
Fixed